app/components/process_list_component.rb

```diff
@@ -20,11 +20,19 @@ def css_class
   end
 
   class ProcessListItemComponent < BaseComponent
-    attr_reader :heading_level, :heading
+    attr_reader :heading_level, :heading, :heading_id
 
-    def initialize(heading_level:, heading:)
+    def initialize(heading_level:, heading:, heading_id: nil)
       @heading_level = heading_level
       @heading = heading
+      @heading_id = heading_id
+    end
+
+    def heading_options
```

Comment:
Not required, but another idea could be to let the component generate its own ID, and expose it to be referenced by markup within it. Could make it a little easier for renderer to avoid magic strings, guarantee unique IDs, and avoid repetition/risk of desync between the ID assignment and reference.

```rb
class ProcessListItemComponent < BaseComponent
  # ...
  def heading_id
    "process-list-item-heading-#{unique_id}"
  end
end
```

```erb
<%= content_tag(heading_level, heading, class: 'usa-process-list__heading', id: heading_id) %>
```

```erb
<%= c.with_item(heading: t('forms.totp_setup.totp_step_1')) do |ic| %>
  <%# ... %>
  <%= render ValidatedFieldComponent.new(
        # ...
        input_html: {
          aria: { labelledby: ic.heading_id },
        },
      ) %>
<% end %>
```

Reply:
>I opted for a constant rather than dynamic ID for easier matching in tests.

That's fair 👍